src/k8s_client/k8s_client.go

```diff
@@ -642,12 +643,12 @@ type taintRemovePatch struct {
 	Path string `json:"path"`
 }
 
-func (c *k8sClient) UntaintNode(name string) error {
+func (c *k8sClient) UninitializedTaintIndex(name string) (int, error) {
```

Comment:
Sorry, I'm not understanding why we need this additional function to find the index, can't the full logic exist as it is in one function?

Reply:
It's now called from both `UntaintNode()` and `waitForInstallationAgentBasedInstaller()`

Reply:
You're right, we don't need to have an additional function to find the index. I'll remove the new function and just use UntaintNode.